internal/olm/client/client_test.go

```diff
@@ -0,0 +1,123 @@
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("check depErrors", func() {
+	Describe("create k8s objects", func() {
+
+		var (
+			fakeClient client.Client
+			csv        olmapiv1alpha1.ClusterServiceVersion
+		)
+		key := types.NamespacedName{
+			Name:      "test.clusterserviceversion.yaml",
+			Namespace: "test-operator-system",
+		}
+		BeforeEach(func() {
+			csv = olmapiv1alpha1.ClusterServiceVersion{
+				Spec: olmapiv1alpha1.ClusterServiceVersionSpec{
+					DisplayName: "test-operator",
+					InstallStrategy: olmapiv1alpha1.NamedInstallStrategy{
+						StrategySpec: olmapiv1alpha1.StrategyDetailsDeployment{
+							DeploymentSpecs: []olmapiv1alpha1.StrategyDeploymentSpec{
+								{
+									Name: "test-operator-controller-manager",
+									Spec: appsv1.DeploymentSpec{
+										Selector: &metav1.LabelSelector{
+											MatchLabels: map[string]string{
+												"control-plane": "controller-manager",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			fakeClient = fake.NewFakeClient(
+				&corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-operator-controller-manager-8687c65f7d-kc44t",
+						Namespace: "test-operator-system",
+						Labels: map[string]string{
+							"control-plane": "controller-manager",
+						},
+					},
+					Status: corev1.PodStatus{
+						Phase: corev1.PodRunning,
+						ContainerStatuses: []corev1.ContainerStatus{
+							{
+								Ready: false,
+								State: corev1.ContainerState{
+									Waiting: &corev1.ContainerStateWaiting{
+										Message: "back-off 5m0s restarting failed container=manager pod=memcached-operator-controller-manager-8687c65f7d-kc44t_memcached-operator-system(d317bedc-7580-4abd-a566-ec80a437bb90)",
+										Reason:  "CrashLoopBackOff",
+									},
+								},
+							},
+						},
+					},
+				},
+				&appsv1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-operator-controller-manager",
+						Namespace: "test-operator-system",
+						Labels: map[string]string{
+							"control-plane": "controller-manager",
+						},
+					},
+					Status: appsv1.DeploymentStatus{
+						Conditions: []appsv1.DeploymentCondition{
+							{
+								Type:   "Available",
+								Status: "False",
+								Reason: "MinimumReplicasUnavailable",
+							},
+							{
+								Type:   "Progressing",
+								Status: "True",
+								Reason: "NewReplicaSetAvailable",
+							},
+						},
+					},
+				},
+			)
+		})
+		Context("with a valid csv", func() {
+			It("should validate the csv successfully", func() {
+				olmclient := Client{KubeClient: fakeClient}
+				err := olmclient.printDeploymentErrors(context.TODO(), key, csv)
+				Expect(err).To(BeNil())
```

Comment:
I would want a test that will also check when the method returns an error. Looks like the only way that will happen is if it can't read a pod.